tools/wptrunner/wptrunner/metadata.py

```diff
@@ -325,8 +326,11 @@ def write_new_expected(metadata_path, expected):
         tmp_path = path + ".tmp"
         try:
             with open(tmp_path, "wb") as f:
-                f.write(manifest_str)
-            os.rename(tmp_path, path)
+                f.write(manifest_str.encode())
+            if sys.version_info >= (3, 3):
+                os.replace(tmp_path, path)
```

Comment:
`os.rename` still exists in Python 3; what was the error that required this?

Reply:
The code expects that `os.rename` will automatically replace the existing file, [which is unfortunately not true on Windows](https://docs.python.org/3.8/library/os.html#os.rename).

>On Windows, if dst exists a FileExistsError is always raised.

>If you want cross-platform overwriting of the destination, use replace().

`os.replace()` is only avaliable on 3.3+ and thus this code.

Reply:
Right, but that is true on Py2.7 as well (https://docs.python.org/2/library/os.html#os.rename), so why is the change now needed for Py3? Is it just an improvement?

Reply:
Ah yes, just an improvement, as I just wanted to remove every error I get. Sorry to confuse you, should I open a separate PR for this?

Reply:
No, that's fine to have in the same PR. Thanks for the explanations :).

Reply:
Fun, somewhat related bug: we want to do the write-to-a-new-file-and-replace thing for the `MANIFEST.json` file, but previously that caused problems on Windows. If you're using Windows and have some time to investigate it would be lovely to fix that. I think it might fix a slew of rare problems where we end up writing a corrupt manifest.

Reply:
@jgraham Is there an existing issue I can read more?

Reply:
https://github.com/web-platform-tests/wpt/issues/9736

Reply:
I tried an experiment and it seems `os.replace` still throws when the destination file is opened.